spec/jobs/risc_delivery_job_spec.rb

```diff
@@ -155,27 +202,49 @@
       end
     end
 
-    context 'slow network errors' do
+    context 'when the job encounters rate limiting' do
       before do
-        stub_request(:post, push_notification_url).to_timeout
+        allow_any_instance_of(described_class).to receive(:analytics).and_return(job_analytics)
+        @redis_rate_limiter = instance_double(RedisRateLimiter)
+        allow(@redis_rate_limiter).to receive(:attempt!).and_raise(RedisRateLimiter::LimitError)
+        allow(RedisRateLimiter).to receive(:new).and_return(@redis_rate_limiter)
       end
 
-      it 'raises and retries via ActiveJob' do
-        expect { perform }.to raise_error(Faraday::ConnectionFailed)
-      end
-
-      context 'it has already failed twice' do
-        before do
-          allow(job).to receive(:executions).and_return 2
+      context 'when the job fails for the 1st time' do
+        it 'raises and retries via ActiveJob' do
+          expect do
+            RiscDeliveryJob.new.perform(
+              push_notification_url: push_notification_url,
+              jwt: jwt,
+              event_type: event_type,
+              issuer: issuer,
+            )
+          end.to raise_error(RedisRateLimiter::LimitError)
+          expect(job_analytics).not_to have_logged_event(
```

Comment:
this is a style nitpick, but can you please leave a newline between multi-line blocks/expectations? i find it much easier to read. thanks!